src/types.ts

```diff
@@ -0,0 +1,25 @@
+export interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+export interface User {
+  id: number;
+  name: string;
+  email: string;
+}
+
+export interface Comment {
+  id: number;
+  postId: number;
+  name: string;
+  email: string;
+  body: string;
+}
+
+export type AllPost = Post & {
+  user?: User;
+  comments?: Comment[];
```

Comment:
Масив завжди у нас буде
```suggestion
  comments: Comment[];
```